File: CS/CustomDrawDayHeader/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using System.IO;
#region #usings
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Drawing;
using System.Drawing.Drawing2D;
#endregion #usings

namespace CustomDrawDayHeader {
    public partial class Form1 : Form {
        const string aptDataFileName = @"..\..\Data\appointments.xml";
        const string resDataFileName = @"..\..\Data\resources.xml";
        
        public Form1() {
            InitializeComponent();
           
           schedulerControl1.Start = new DateTime(2008, 7, 11);
           FillData();
        }

        #region FillData
        void FillData() {
            AppointmentCustomFieldMapping customNameMapping = new AppointmentCustomFieldMapping("CustomName", "CustomName");
            AppointmentCustomFieldMapping customStatusMapping = new AppointmentCustomFieldMapping("CustomStatus", "CustomStatus");
            schedulerStorage1.Appointments.CustomFieldMappings.Add(customNameMapping);
            schedulerStorage1.Appointments.CustomFieldMappings.Add(customStatusMapping);
            FillResourcesStorage(schedulerStorage1.Resources.Items, resDataFileName);
            FillAppointmentsStorage(schedulerStorage1.Appointments.Items, aptDataFileName);
        }

        static Stream GetFileStream(string fileName) {
            return new StreamReader(fileName).BaseStream;
        }

        static void FillAppointmentsStorage(AppointmentCollection c, string fileName) {
            using (Stream stream = GetFileStream(fileName)) {
                c.ReadXml(stream);
                stream.Close();
            }
        }

        static void FillResourcesStorage(ResourceCollection c, string fileName) {
            using (Stream stream = GetFileStream(fileName)) {
                c.ReadXml(stream);
                stream.Close();
            }
        }
        #endregion

        private void schedulerStorage_AppointmentsChanged(object sender, PersistentObjectsEventArgs e) {
            schedulerStorage1.Appointments.Items.WriteXml(aptDataFileName);
        }

        private void cb_CustomDraw_CheckedChanged(object sender, EventArgs e) {
            if(cb_CustomDraw.Checked) {
                schedulerControl1.CustomDrawDayHeader += new DevExpress.XtraScheduler.CustomDrawObjectEventHandler(schedulerControl1_CustomDrawDayHeader);
            }
            else {
                schedulerControl1.CustomDrawDayHeader -= new DevExpress.XtraScheduler.CustomDrawObjectEventHandler(schedulerControl1_CustomDrawDayHeader);
            }
            schedulerControl1.Refresh();

        }
#region #customdrawdayheader
private void schedulerControl1_CustomDrawDayHeader(object sender, CustomDrawObjectEventArgs e) {
    DayHeader header = e.ObjectInfo as DayHeader;
    // Draws the outer rectangle.
    e.Cache.FillRectangle(new LinearGradientBrush(e.Bounds,
        Color.LightBlue, Color.Blue, LinearGradientMode.Vertical), e.Bounds);
    Rectangle innerRect = Rectangle.Inflate(e.Bounds, -2, -2);
    // Draws the inner rectangle.
    e.Cache.FillRectangle(new LinearGradientBrush(e.Bounds,
        Color.Blue, Color.LightSkyBlue, LinearGradientMode.Vertical), innerRect);
    // Draws the header's caption.
    e.Cache.DrawString(header.Caption, header.Appearance.HeaderCaption.Font,
        new SolidBrush(Color.White), innerRect,
        header.Appearance.HeaderCaption.GetStringFormat());
    e.Handled = true;
}
#endregion #customdrawdayheader
    }
}

File: VB/CustomDrawDayHeader/Form1.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports System.IO
#Region "#usings"
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Drawing
Imports System.Drawing.Drawing2D
#End Region ' #usings

Namespace CustomDrawDayHeader
    Partial Public Class Form1
        Inherits Form

        Private Const aptDataFileName As String = "..\..\Data\appointments.xml"
        Private Const resDataFileName As String = "..\..\Data\resources.xml"

        Public Sub New()
            InitializeComponent()

           schedulerControl1.Start = New Date(2008, 7, 11)
           FillData()
        End Sub

        #Region "FillData"
        Private Sub FillData()
            Dim customNameMapping As New AppointmentCustomFieldMapping("CustomName", "CustomName")
            Dim customStatusMapping As New AppointmentCustomFieldMapping("CustomStatus", "CustomStatus")
            schedulerStorage1.Appointments.CustomFieldMappings.Add(customNameMapping)
            schedulerStorage1.Appointments.CustomFieldMappings.Add(customStatusMapping)
            FillResourcesStorage(schedulerStorage1.Resources.Items, resDataFileName)
            FillAppointmentsStorage(schedulerStorage1.Appointments.Items, aptDataFileName)
        End Sub

        Private Shared Function GetFileStream(ByVal fileName As String) As Stream
            Return (New StreamReader(fileName)).BaseStream
        End Function

        Private Shared Sub FillAppointmentsStorage(ByVal c As AppointmentCollection, ByVal fileName As String)
            Using stream As Stream = GetFileStream(fileName)
                c.ReadXml(stream)
                stream.Close()
            End Using
        End Sub

        Private Shared Sub FillResourcesStorage(ByVal c As ResourceCollection, ByVal fileName As String)
            Using stream As Stream = GetFileStream(fileName)
                c.ReadXml(stream)
                stream.Close()
            End Using
        End Sub
        #End Region

        Private Sub schedulerStorage_AppointmentsChanged(ByVal sender As Object, ByVal e As PersistentObjectsEventArgs) Handles schedulerStorage1.AppointmentsChanged, schedulerStorage1.AppointmentsInserted, schedulerStorage1.AppointmentsDeleted
            schedulerStorage1.Appointments.Items.WriteXml(aptDataFileName)
        End Sub

        Private Sub cb_CustomDraw_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cb_CustomDraw.CheckedChanged
            If cb_CustomDraw.Checked Then
                AddHandler schedulerControl1.CustomDrawDayHeader, AddressOf schedulerControl1_CustomDrawDayHeader
            Else
                RemoveHandler schedulerControl1.CustomDrawDayHeader, AddressOf schedulerControl1_CustomDrawDayHeader
            End If
            schedulerControl1.Refresh()

        End Sub
#Region "#customdrawdayheader"
Private Sub schedulerControl1_CustomDrawDayHeader(ByVal sender As Object, ByVal e As CustomDrawObjectEventArgs)
    Dim header As DayHeader = TryCast(e.ObjectInfo, DayHeader)
    ' Draws the outer rectangle.
    e.Cache.FillRectangle(New LinearGradientBrush(e.Bounds, Color.LightBlue, Color.Blue, LinearGradientMode.Vertical), e.Bounds)
    Dim innerRect As Rectangle = Rectangle.Inflate(e.Bounds, -2, -2)
    ' Draws the inner rectangle.
    e.Cache.FillRectangle(New LinearGradientBrush(e.Bounds, Color.Blue, Color.LightSkyBlue, LinearGradientMode.Vertical), innerRect)
    ' Draws the header's caption.
    e.Cache.DrawString(header.Caption, header.Appearance.HeaderCaption.Font, New SolidBrush(Color.White), innerRect, header.Appearance.HeaderCaption.GetStringFormat())
    e.Handled = True
End Sub
#End Region ' #customdrawdayheader
    End Class
End Namespace